homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,489 @@
+
+from aioswitcher.schedules import SwitcherV2Schedule  # noqa F401
+from aioswitcher.consts import (MONDAY, TUESDAY, WEDNESDAY, THURSDAY,
+                                FRIDAY, SATURDAY, SUNDAY)
+
+from homeassistant.core import HomeAssistant, Event, ServiceCall
+from homeassistant.const import (CONF_FRIENDLY_NAME, CONF_ICON, CONF_NAME,
+                                 EVENT_HOMEASSISTANT_STOP, SERVICE_TURN_ON,
+                                 SERVICE_TURN_OFF)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import (DOMAIN as SWITCH_DOMAIN,
+                                             SwitchDevice)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import (async_load_platform,
+                                             async_listen_platform)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+
+REQUIREMENTS = ['aioswitcher==2019.2.6']
+
+_LOGGER = getLogger(__name__)
+
+DOMAIN = 'switcher_kis'
+ENTITY_ID_FORMAT = DOMAIN + '_{}'
+
+CONF_PHONE_ID = 'phone_id'
+CONF_DEVICE_ID = 'device_id'
+CONF_DEVICE_PASSWORD = 'device_password'
+CONF_INCLUDE_SCHEDULE_SENSORS = 'include_schedule_sensors'
+CONF_SCHEDULE_SCAN_INTERVAL = 'schedules_scan_interval'
+
+CONF_AUTO_OFF = 'auto_off'
+CONF_SCHEDULE_ID = 'schedule_id'
+CONF_RECURRING = 'recurring'
+CONF_START_TIME = 'start_time'
+CONF_END_TIME = 'end_time'
+CONF_DAYS = 'days'
+
+DISCOVERY_CONFIG = 'config'
+DISCOVERY_DEVICE = 'device'
+DISCOVERY_SCHEDULES = 'schedules'
+
+DEFAULT_NAME = 'boiler'
+DEFAULT_SCHEDULES_SCAN_INTERVAL = timedelta(minutes=5)
+DEFAULT_CONF_DAYS = []  # type: List[str]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PHONE_ID): cv.string,
+        vol.Required(CONF_DEVICE_ID): cv.string,
+        vol.Required(CONF_DEVICE_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME,
+                     default=DEFAULT_NAME): cv.slugify,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+        vol.Optional(CONF_ICON): cv.icon,
+        vol.Optional(CONF_INCLUDE_SCHEDULE_SENSORS,
+                     default=False): cv.boolean,
+        vol.Optional(CONF_SCHEDULE_SCAN_INTERVAL,
+                     default=DEFAULT_SCHEDULES_SCAN_INTERVAL): vol.All(
+                         cv.time_period, cv.positive_timedelta)
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SERVICE_TURN_ON_15 = 'turn_on_15_minutes'
+SERVICE_TURN_ON_30 = 'turn_on_30_minutes'
+SERVICE_TURN_ON_45 = 'turn_on_45_minutes'
+SERVICE_TURN_ON_60 = 'turn_on_60_minutes'
+
+SERVICE_SET_AUTO_OFF = 'set_auto_off'
+SET_AUTO_OFF_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_AUTO_OFF): cv.time_period_str
+})
+
+SERVICE_UPDATE_DEVICE_NAME = 'update_device_name'
+UPDATE_DEVICE_NAME_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): vol.All(
+        cv.string, vol.Length(min=2, max=32))
+})
+
+
```

Comment:
Move the following part to service_registration.py

Reply:
Made the required changes [here](https://github.com/home-assistant/home-assistant/pull/20830/commits/a67dcbfefe43892627db2fc64162a658f5b89d45).
Thank you.